pkg/crc/gpg/gpg.go

```diff
@@ -0,0 +1,102 @@
+package gpg
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/keybase/go-crypto/openpgp/armor"
+	"github.com/keybase/go-crypto/openpgp/packet"
+)
+
+func getSignature(signatureFilePath string) (*packet.Signature, error) {
+	sigFile, err := os.Open(signatureFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := sigFile.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	sigFileBlock, err := armor.Decode(sigFile)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding signature file: %w", err)
+	}
+	if sigFileBlock.Type != "PGP SIGNATURE" {
+		return nil, errors.New("not an armored signature")
+	}
+
+	sigFileBody, err := packet.Read(sigFileBlock.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading signature file: %w", err)
+	}
+	signature, ok := sigFileBody.(*packet.Signature)
+	if !ok {
+		return nil, errors.New("not a valid signature file")
+	}
+	return signature, nil
+}
+
+func getPubicKey() (*packet.PublicKey, error) {
+	pubKeyBlock, err := armor.Decode(bytes.NewReader([]byte(constants.GPGPublicKey)))
+	if err != nil {
+		return nil, fmt.Errorf("error decoding public key: %s", err)
+	}
+	if pubKeyBlock.Type != "PGP PUBLIC KEY BLOCK" {
```

Comment:
there is no constant defined in the library for this?

Reply:
@gbraad No it is not I did check that first :)

Reply:
very weird. very prone to error